meson.build

```diff
@@ -857,6 +857,13 @@ endif
 
 
 #####################################################################
+if get_option('cross-tools-only') != true
```

Comment:
This warrants a comment, for those unfamiliar with cross-builds requirements

Reply:
I havn't touched this PR in more than a year. The need for cross-tools only is still there, but the way systemd handles build makes it quite complicated and i'm not sure my approch is the right one...

Reply:
Yeah looking at the backlog.

The approach is a tad invasive, but I cannot think of a better way to handle it. And I understand the need for this, as I work daily with Yocto.

At the very minimum though, an automated build test is needed, to ensure the behaviour you want doesn't bitrot.

Reply:
heh, that one was originally thought for buildroot... I tried to work around the way systemd handles $prefix and the assumption that systemd can only be installed in the root directory (which of course isn't true when you install only cross-tools)

I think fixing that assumption first would be the way to go, but it's really deeply ingrained, and I have very little meson skill...

Reply:
What do you mean by that? The prefix of most things can be changed with -Dprefix and -Drootprefix - only /etc/init.d/README, /usr/share/bash-completion and /var are ignoring it. zsh is installed in the right place, so looks like bash-completion should be fixable.

Reply:
The following diff ensures everything is installed under the prefix and rootprefix meson variables - and I think it's even too aggressive, as some of those you'd just disable for minimal tools builds:

```
diff --git a/docs/var-log/meson.build b/docs/var-log/meson.build
index ba7957711d..b3530f73fe 100644
--- a/docs/var-log/meson.build
+++ b/docs/var-log/meson.build
@@ -2,5 +2,5 @@
 
 if conf.get('HAVE_SYSV_COMPAT') == 1 and get_option('create-log-dirs')
         install_data('README',
-                     install_dir : '/var/log')
+                     install_dir : join_paths(prefixdir, 'var/log'))
 endif
diff --git a/meson.build b/meson.build
index 82c6de900c..77862fc347 100644
--- a/meson.build
+++ b/meson.build
@@ -90,11 +90,6 @@ rootprefixdir_noslash = rootprefixdir == '/' ? '' : rootprefixdir
 
 have_standalone_binaries = get_option('standalone-binaries')
 
-sysvinit_path = get_option('sysvinit-path')
-sysvrcnd_path = get_option('sysvrcnd-path')
-conf.set10('HAVE_SYSV_COMPAT', sysvinit_path != '' and sysvrcnd_path != '',
-           description : 'SysV init scripts and rcN.d links are supported')
-
 if get_option('hibernate') and not get_option('initrd')
         error('hibernate depends on initrd')
 endif
@@ -110,12 +105,18 @@ prefixdir = get_option('prefix')
 if not prefixdir.startswith('/')
         error('Prefix is not absolute: "@0@"'.format(prefixdir))
 endif
+
+sysvinit_path = join_paths(prefixdir, get_option('sysvinit-path'))
+sysvrcnd_path = join_paths(prefixdir, get_option('sysvrcnd-path'))
+conf.set10('HAVE_SYSV_COMPAT', sysvinit_path != prefixdir and sysvrcnd_path != prefixdir,
+           description : 'SysV init scripts and rcN.d links are supported')
+
 bindir = join_paths(prefixdir, get_option('bindir'))
 libdir = join_paths(prefixdir, get_option('libdir'))
 sysconfdir = join_paths(prefixdir, get_option('sysconfdir'))
 includedir = join_paths(prefixdir, get_option('includedir'))
 datadir = join_paths(prefixdir, get_option('datadir'))
-localstatedir = join_paths('/', get_option('localstatedir'))
+localstatedir = join_paths(prefixdir, get_option('localstatedir'))
 
 rootbindir = join_paths(rootprefixdir, 'bin')
 rootsbindir = join_paths(rootprefixdir, split_bin ? 'sbin' : 'bin')
diff --git a/meson_options.txt b/meson_options.txt
index eed3596b9b..71beab0df3 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -32,9 +32,9 @@ option('static-libudev', type : 'combo',
 option('standalone-binaries', type : 'boolean', value : 'false',
        description : 'also build standalone versions of supported binaries')
 
-option('sysvinit-path', type : 'string', value : '/etc/init.d',
+option('sysvinit-path', type : 'string', value : 'etc/init.d',
        description : 'the directory where the SysV init scripts are located')
-option('sysvrcnd-path', type : 'string', value : '/etc/rc.d',
+option('sysvrcnd-path', type : 'string', value : 'etc/rc.d',
        description : 'the base directory for SysV rcN.d directories')
 option('telinit-path', type : 'string', value : '/lib/sysvinit/telinit',
        description : 'path to telinit')
diff --git a/shell-completion/bash/meson.build b/shell-completion/bash/meson.build
index 1afa7977e6..efc63e250c 100644
--- a/shell-completion/bash/meson.build
+++ b/shell-completion/bash/meson.build
@@ -4,7 +4,7 @@ bashcompletiondir = get_option('bashcompletiondir')
 if bashcompletiondir == ''
         bash_completion = dependency('bash-completion', required : false)
         if bash_completion.found()
-                bashcompletiondir = bash_completion.get_pkgconfig_variable('completionsdir')
+                bashcompletiondir = prefixdir + bash_completion.get_pkgconfig_variable('completionsdir')
         else
                 bashcompletiondir = join_paths(datadir, 'bash-completion/completions')
         endif
diff --git a/src/journal-remote/meson.build b/src/journal-remote/meson.build
index 4572f4bd1c..7f9218be2d 100644
--- a/src/journal-remote/meson.build
+++ b/src/journal-remote/meson.build
@@ -68,10 +68,11 @@ if conf.get('ENABLE_REMOTE') == 1 and conf.get('HAVE_MICROHTTPD') == 1
                      install_dir : join_paths(pkgdatadir, 'gatewayd'))
 
         if get_option('create-log-dirs')
+                remotedir = join_paths(prefixdir, 'var/log/journal/remote')
                 meson.add_install_script('sh', '-c',
-                                         mkdir_p.format('/var/log/journal/remote'))
+                                         mkdir_p.format(remotedir))
                 meson.add_install_script('sh', '-c',
-                                         '''chown 0:0 $DESTDIR/var/log/journal/remote &&
-                                            chmod 755 $DESTDIR/var/log/journal/remote || :''')
+                                         '''chown 0:0 $DESTDIR''' + remotedir + ''' &&
+                                            chmod 755 $DESTDIR''' + remotedir + ''' || :''')
         endif
 endif
diff --git a/src/journal/meson.build b/src/journal/meson.build
index 7aea28d129..b5f963456d 100644
--- a/src/journal/meson.build
+++ b/src/journal/meson.build
@@ -113,21 +113,22 @@ if install_sysconfdir
 endif
 
 if get_option('create-log-dirs')
+        journaldir = join_paths(prefixdir, 'var/log/journal')
         meson.add_install_script(
                 'sh', '-c',
-                mkdir_p.format('/var/log/journal'))
+                mkdir_p.format(journaldir))
         meson.add_install_script(
                 'sh', '-c',
-                '''chown 0:0 $DESTDIR/var/log/journal &&
-                   chmod 755 $DESTDIR/var/log/journal || :''')
+                '''chown 0:0 $DESTDIR''' + journaldir + ''' &&
+                   chmod 755 $DESTDIR''' + journaldir + ''' || :''')
         if get_option('adm-group')
                 meson.add_install_script(
                         'sh', '-c',
-                        'setfacl -nm g:adm:rx,d:g:adm:rx $DESTDIR/var/log/journal || :')
+                        'setfacl -nm g:adm:rx,d:g:adm:rx $DESTDIR' + journaldir + ' || :')
         endif
         if get_option('wheel-group')
                 meson.add_install_script(
                         'sh', '-c',
-                        'setfacl -nm g:wheel:rx,d:g:wheel:rx $DESTDIR/var/log/journal || :')
+                        'setfacl -nm g:wheel:rx,d:g:wheel:rx $DESTDIR' + journaldir + ' || :')
         endif
 endif
```